sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplate.java

```diff
@@ -465,6 +469,44 @@ public <T> Mono<T> insert(String containerName, T objectToSave) {
         return insert(containerName, objectToSave, null);
     }
 
+    /**
+     * Insert all items with bulk.
+     *
+     * @param entityInformation the CosmosEntityInformation
+     * @param entities the Iterable entities to be deleted
+     * @param <T> type class of domain type
+     * @param <S> type class of domain type
+     * @return Flux of result
+     */
+    public <S extends T, T> Flux<S> insertAll(CosmosEntityInformation<T, ?> entityInformation, Iterable<S> entities) {
```

Comment:
For reactive version, we should ideally take both Iterable and Publisher, so we can avoid the double conversion happening in SimpleReactiveCosmosRepository and ReactiveCosmosTemplate -> 


`return cosmosOperations.insertAll(this.entityInformation, Flux.from(entityStream).toIterable());`

`.executeBulkOperations(Flux.fromIterable(cosmosItemOperations))`